aimmo-game/simulation/location.py

```diff
@@ -1,10 +1,24 @@
 class Location(object):
     def __init__(self, x, y):
-        self.x = x
-        self.y = y
+        self._x = x
+        self._y = y
+
+    @property
+    def x(self):
+        return self._x
+
+    @property
+    def y(self):
+        return self._y
 
     def __add__(self, direction):
-        return Location(self.x + direction.x, self.y + direction.y)
+        try:
+            d_x = direction.x
+            d_y = direction.y
+        except AttributeError:
+            return self
```

Comment:
I don't like the fact that adding a Location to something nonsensical (eg an integer) will fail silently.